src/utils/ticket/YouTrackTicketStatusFetcher.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace staabm\PHPStanTodoBy\utils\ticket;
+
+use RuntimeException;
+use function array_key_exists;
+use function is_array;
+use function is_null;
+use function is_string;
+
+final class YouTrackTicketStatusFetcher implements TicketStatusFetcher
+{
+    private string $host;
+    private string $authorizationHeader;
+
+    /**
+     * @var array<string, ?string>
+     */
+    private array $cache;
+
+    public function __construct(string $host, ?string $credentials, ?string $credentialsFilePath)
+    {
+        $credentials = self::getCredentials($credentials, $credentialsFilePath);
+
+        $this->host = $host;
+        $this->authorizationHeader = self::createAuthorizationHeader($credentials);
+
+        $this->cache = [];
+    }
+
+    public function fetchTicketStatus(string $ticketKey): ?string
+    {
+        if (array_key_exists($ticketKey, $this->cache)) {
+            return $this->cache[$ticketKey];
+        }
+
+        $curl = curl_init("{$this->host}/api/issues/$ticketKey?fields=resolved");
```

Comment:
I recently added a HttpClient into the other fetcher. try to use that

Reply:
I also noticed this one and applied it just now.